lighthouse-core/config/config.js

```diff
@@ -160,18 +160,18 @@ function assertValidGatherer(gathererInstance, gathererName) {
 /**
  * Creates a settings object from potential flags object by dropping all the properties
  * that don't exist on Config.Settings.
- * TODO(bckenny): fix Flags type
  * @param {Partial<LH.Flags>=} flags
- * @return {Partial<LH.Config.Settings>}
- */
+ * @return {RecursivePartial<LH.Config.Settings>}
```

Comment:
It's a bug that this wasn't flagged before, but it wasn't until I made any change to `LH.Flags.output` that it started to be flagged (even though the problem is with `LH.Flags.throttling`, weirdly enough).

We just want to copy any `Settings` properties over from an instance of `Flags`, so the result will only have `Settings` properties but some may not be set. So `Partial<LH.Config.Settings>` would be correct if `Settings` were a simple object.

The issue is that one of the properties (`throttling`) is itself an object, and while `Flags.throttling` has all optional properties, on `Settings.throttling` they're all required. This change applies `Partial<>` to all of `Settings` properties recursively. It could special case `throttling` instead, but this will catch any future objects added to `SharedFlagsSettings` as well.